### PR TITLE
chore: add fuzz unit test and fix a few fuzz edge cases

### PR DESCRIPTION
**Pull request by justinsb**

### CODE REVIEW

File: pkg/test/fuzz/generate.go [hunk 1]
```diff
@@ -67,8 +67,8 @@ func fillWithRandom0(t *testing.T, randStream *rand.Rand, msg protoreflect.Messa
 		field := fields.Get(i)
 
 		if field.IsList() {
-			count := randStream.Intn(10)
-			// Bias heavily to zero
+			count := randStream.Intn(20)
+			// Bias heavily to zero; we don't want to create infinitely large messages
```
Comment (cheftako) [inline pkg/test/fuzz/generate.go:71]: Would it make sense to find a more curved distribution system? 0 seems to just not modify the field at all. It would seem like something like 60% 0, 25% 1, 10% 2,  4% v3 and 1% 4, might give better testing while helping with message size?
